internal/devpkg/narinfo_cache.go

```diff
@@ -25,6 +25,10 @@ const BinaryCache = "https://cache.nixos.org"
 // ALERT: Callers in a perf-sensitive code path should call FillNarInfoCache
 // before calling this function.
 func (p *Package) IsInBinaryCache() (bool, error) {
+	// Patched glibc packages are not in the binary cache.
+	if p.PatchGlibc {
```

Comment:
this should be `PatchGlibc && linux`

or better still the PatchGlibc boolean should only be set if it is specified in the devbox.json _and_ the current OS is linux